[REFACTOR] MVC 컨트롤러 리팩토링과 커스텀 템플릿 개발

yjy323 opened the pull request:
## 🪺 개요 
엇.. 템플릿 디렉토리를 별도로 업로드 했어야 할 것 같네요,, 수정된 파일이 763개일줄은,,,,, 몰랐습니다 죄송해요
코드 변경 사항이랑 템플릿 작업 시 주의 사항 위주로 구체적으로 작성하겠습니다!

## 💻 작업 사항 
### JWT 인증
- 로그인 및 회원가입 기능을 추가하면서 Security Config 설정을 수정했습니다.
- 로그인 하지 않을 시, 403 에러가 발생할 수 있습니다.
- Swagger 사용 시에는 403 에러가 발생하지 않도록 Config 설정을 추가했습니다.
  - 해결 방법: 회원가입 → 로그인 → Response body의 토큰 값 복사 → Swagger UI 상단의 Authorize 활성화

### MVC 컨트롤러 리팩토링
- 기존 be 패키지는 그대로 두고, full 패키지를 추가해서 리팩토링 작업을 수행했습니다.
  - 추후 be 패키지 명을 `/api/v1`로 수정해도 좋을 것 같습니다.
- RESTful API 서버에서 Spring MVC 서버로 리팩토링하기 위해 @RestController 어노테이션을 @Controller로 수정했습니다.
- Controller를 수정하면서 ApiResponse를 응답하는 대신 템플릿 View를 반환하도록 수정했습니다.
- Controller 내부에 Model 객체를 추가하고, POST API에는 BindingResult 객체를 추가해 에러 UI를 추가했습니다.
- 스프링 MVC의 @ModelAttribute는 기본적으로 Java Bean 규약에 따른 Setter 메서드를 통해 데이터를 바인딩하므로 기존 Request DTO에 @Setter 어노테이션을 추가했습니다.
- JWT 토큰 정보가 요청에 담겨있다면 index.html로, 없다면 login.html로 리다이렉션하기 위해 HomeController를 추가했습니다.

### 커스텀 템플릿 개발
- INSPINA 템플릿의 여러 컴포넌트를 재사용해 다음의 화면을 개발했습니다.
- [x] 로그인 및 회원가입
- [x] 메인 페이지 및 navbar, topbar
- [x] 상담 예약
- [x] 예약된 상담 목록 조회
- [x] 상담 예약 상세 조회
- [x] 상담 예약 승인/거절
- [x] 상담 시작
- [x] 상담 목록 조회
- [x] 상담 노트 작성/수정/추가
- [x] 상담 완료

### 템플릿 화면
- 회원가입 시 DB에 유저 정보가 저장됩니다.
- 해당 유저 정보로 로그인 할 수 있습니다.
- 상담 예약 → 예약 목록 조회 → 예약 상세 조회 → 승인/거절 → 상담 시작 → 상담 노트 작성 → 상담 완료 까지 브라우저에서 테스트할 수 있습니다.
- ~~상담 완료 → 미배정 사건 생성까지 처리되지만, 해당 PR에서는 화면이 구현되지 않았습니다.~~
  - **DB에는 저장된 것으로 확인했습니다.**
  
<img width="830" height="79" alt="image" src="https://github.com/user-attachments/assets/37b56462-625d-48f7-8856-c5c33fad997c" />


**TODO**
- [ ] 파일 업로드 기능 추가
- [ ] ~~사건 목록 조회(Swagger를 통해 `상담 완료 시 사건 생성`까지는 확인했습니다.)~~ **Cases Domain에서 관련 에러 발생**

### AI 사용 시 주의점
- AI가 템플릿 컴포넌트를 재사용하지 않고, 임의로 작성하는 할루시네이션이 자주 발생합니다. 개발하려는 화면과 비슷한 기존 템플릿의 코드를 추가로 제공하거나, 상담 템플릿을 레퍼런스로 제공하면 좋습니다.
- 절대 임의로 템플릿을 작성하지 말고, 컴포넌트를 확인해서 재사용하라는 지침이 프롬프트에 필요합니다.
- 입력 Form, 파일 업로드 등 템플릿의 특정 요소를 활용하려면 그 파일을 정확히 지정해서 지침을 작성할 필요가 있습니다. 그래서 템플릿 작업을 하기 전에, 어떤 파일에 어떤 요소가 있는지 알고 사용해야 품질이 좋습니다.
  - Ex) 상담 노트 작성 화면 개발 시, `@.../templates/form/validation.html을 참고해`

## 🌱 Issue Number
- #44

## ✅ 셀프 체크리스트
- [x] PR 제목을 형식에 맞게 작성했나요 ?
- [x] 브랜치 전략에 맞는 브랜치에 PR을 올리고 있나요 ?
- [x] 테스트는 잘 통과했나요 ?
- [x] 빌드에 성공했나요 ?
- [x] 본인을 Assign 해주세요.
- [x] 리뷰할 팀원들을 Request 해주세요.
- [x] 해당 PR에 맞는 label을 붙여주세요.

## 🙏 To Reviewers
> @RestController에서 @Controller로 수정하고, BindingResult 객체와 Model 객체를 추가했습니다. 이 부분에 오류가 있는지만 리뷰 해주셔도 될 것 같아요! 나머지 파일은 템플릿 파일들입니다!

yjy323 committed:
feat: 풀스택 MVC 애플리케이션 구조 추가

- Thymeleaf 템플릿 엔진 의존성 추가
- com.haeil.full 패키지에 전체 도메인 복사
- 정적 리소스 및 템플릿 디렉토리 생성
- FullApplication 메인 클래스 설정

#44

yjy323 committed:
style: 코드 포매팅 및 스타일 정리

- import 문 순서 정리
- 주석 스타일 통일 (// 형식으로 변경)
- 파라미터 줄바꿈 스타일 정리
- 코드 가독성 향상

#44

yjy323 committed:
feat: 상담 관리 웹 인터페이스 구현

- ConsultationController에 웹 페이지 렌더링 메서드 추가
- 상담 예약 등록/조회 템플릿 구현
- 상담 진행/완료 관리 템플릿 구현
- 유효성 검증 및 에러 핸들링 추가
- 사이드 내비게이션에 상담 메뉴 연결

#44

yjy323 committed:
feat: 상담 관리 기능 개선 및 UI 향상

- 예약에서 상담 시작으로 연결하는 워크플로우 구현
- 상담 상세 페이지에 상담 기록 관리 기능 추가
- 상담 기록 생성/수정 모달 구현
- 예약 상세 페이지에서 상담 시작 버튼 추가
- 클라이언트 요청 DTO 유효성 검증 강화
- 사이드 내비게이션 메뉴 구조 개선

#44

yjy323 reviewed (commented):
파일 하나씩 넘겨가면서 모두 리뷰하실 필요는 없습니다..! 옆에 내비 바에서 아래 파일만 찾아서 리뷰해주세요!
1. src에서는 full 패키지 하위의 Controller 파일
2. src/main/resources/templates 하위의 projects/consultation/ 하위의 html 파일

templates 파일은 사실 리뷰보다는 템플릿을 구매하면 이런 식으로 개발할 수 있구나 의 한 예시로 흝어만 보셔도 좋을 것 같습니다!

yjy323 committed:
fix: Swagger 보안 설정 강화

- BE와 Full 환경 SwaggerConfig에 보안 요구사항 추가 설정
- API 보안 인증 흐름 개선

#44

erika0915 reviewed (approved):
코드를 하나하나 확인하는 것보다 일단 실행시켜서 상담 부분이 잘 동작하는지 확인 완료 했습니당 ~ template으로 하는건 처음이라서 동작되는거 보니깐 신기하네요 ~ ㅎㅎ👍🏻 

하나 말씀드리면, SecurityConfig에 그 consultation 부분은 role이 SECRETARY와 ATTORNEY만 되는 것 같은데, role로 COUNSEL을 부여해줘야할 것 같아요 ~ 상담 부분 하셨다고 해서, 처음 회원가입 때 상담 변호사로 들어갔는데 403이 뜨서 보니깐 SecurityConfig에 COUNSEL이 안되어 있어서용~ !! 

그리고 궁금한거는, 그러면 앞으로 개발을 full 패키지에서 mvc 방식으로 템플릿 기반으로 진행하면 될까요 ? 아니면 기존처럼 be 패키지에서 json api 형태로 개발하는게 맞을까요 ?

너무너무~~~고생하셨습니다 !!!! 💯🥇

Chominsung0823 reviewed (approved):
실행해서 모두 확인했습니다! 제가 이어받아서 full 폴더에서 작업하면 되는 거죠?!
고생하셨습니다!! 👍

yjy323 commented:
> 코드를 하나하나 확인하는 것보다 일단 실행시켜서 상담 부분이 잘 동작하는지 확인 완료 했습니당 ~ template으로 하는건 처음이라서 동작되는거 보니깐 신기하네요 ~ ㅎㅎ👍🏻
> 
> 하나 말씀드리면, SecurityConfig에 그 consultation 부분은 role이 SECRETARY와 ATTORNEY만 되는 것 같은데, role로 COUNSEL을 부여해줘야할 것 같아요 ~ 상담 부분 하셨다고 해서, 처음 회원가입 때 상담 변호사로 들어갔는데 403이 뜨서 보니깐 SecurityConfig에 COUNSEL이 안되어 있어서용~ !!
> 
> 그리고 궁금한거는, 그러면 앞으로 개발을 full 패키지에서 mvc 방식으로 템플릿 기반으로 진행하면 될까요 ? 아니면 기존처럼 be 패키지에서 json api 형태로 개발하는게 맞을까요 ?
> 
> 너무너무~~~고생하셨습니다 !!!! 💯🥇

헉 SecurityConfig 확인하고 수정하겠습니다!
당장 프로젝트 완성하는데는 템플릿 기반으로 개발해야 할 것 같습니다! 그리고 Controller 외에 다른 레이어는 동일하게 개발해도 괜찮습니다

yjy323 commented:
> 실행해서 모두 확인했습니다! 제가 이어받아서 full 폴더에서 작업하면 되는 거죠?! 고생하셨습니다!! 👍

네네 맞습니다! 수희님 리뷰대로 SpringConfig만 수정하고 머지하겠습니다.

yjy323 committed:
feat: 권한 설정 개선 및 관리자 접근 권한 추가 #44

- 미배정 사건 접근 권한에 ADMIN 역할 추가
- 요청/진행/완료 사건 접근 권한에 ADMIN 역할 추가
- 상담 관련 API에 COUNSEL, ADMIN 역할 추가
- 불필요한 HttpMethod 임포트 제거
- 코드 포매팅 개선

yjy323 commented:
수희님 리뷰대로 SecurityConfing에 상담 변호사 ROLE 추가하고, 서버 올려서 회원가입 - 로그인 - 상담 기능 테스트 완료했습니다~~
원래 상담 변호사로 로그인을 하면 상담 목록을 조회할 수 없었던 문제였던거죠?!
